### PR TITLE
disable contiguity on cross dimensional overlapped tensor

### PR DESCRIPTION
**Pull request by jjsjann123**
Unmarked contiguity on stride properties when we have dimensions potentially covering overlapping memory.
This check could be done more accurately, per dimension instead of a global flag per tensor. I'm just keeping it simple here, as the existing code gives us correctness and that's what's important.

### CODE REVIEW

**Comment by jjsjann123** [comment]
For some context, this PR comes from the discussion here: https://github.com/csarofeen/pytorch/pull/1519#discussion_r827278412

File: aten/src/ATen/core/tensor_type.cpp [hunk 1]
```diff
@@ -173,21 +211,35 @@ VaryingShape<Stride> TensorType::computeStrideProps(
         }
       }
     }
+    // conveniently is_contiguous_strides/is_contiguous_strides only returns
+    // true when there's no memory overlap, so we only re-compute has_overlap
+    // in the last branch when both returns false
+    if (!tensor_contiguity) {
+      // trust tensor_contiguity and only computes overlap when it is not set
+      has_overlap = possible_cross_dimension_overlap(sizes, strides);
```
Comment (ngimel) [inline aten/src/ATen/core/tensor_type.cpp:219]: you should be computing overlap only if checks for `contiguous` and `channels_last` in lines 172 and 177 weren't true

Comment (jjsjann123) [reply aten/src/ATen/core/tensor_type.cpp:219]: That's what's happening since this line is inside the false block of line 180.

Comment (ngimel) [reply aten/src/ATen/core/tensor_type.cpp:219]: Oh, right, sorry I missed that

**Review by ngimel — APPROVED** [review]

**Comment by malfet** [comment]
Have to revert this one, as it conflicts with #74520, which were landed using an old "import to fbsource" mechanism

**Review by malfet — APPROVED** [review]

**Comment by jjsjann123** [comment]
diff looks good to me~

**Pull request closed without merging**